### DISCUSSION TITLE
How to dynamically close context menu after preventDefault on Menu Item

### DISCUSSION THREAD

**Discussion by flexa-david** [radix-ui/primitives#1447] [Help] [answered]
I have a grid, where right clicking on each cell brings up a Context Menu. When I click the MenuItem that adds a column to the right, I want it to focus on the newly created cell. I tried preventDefault() on the MenuItem select handler in order to stop it focusing on the origin cell, however I don't know how to close the menu dynamically after that. 

Also attempting to dynamically focus on the newly created cell element doesn't seem to be working.

Any ideas, at least on how to close the ContextMenu dynamically?

**Comment by benoitgrelard** [comment, marked as answer]
Hey @flexa-david,

Not sure if I fully understood what you're trying to achieve, but here's a quick and dirty demo of how you could achieve it (from my understanding)

https://codesandbox.io/s/contextmenu-table-focus-xpk9fi?file=/src/App.js

**Reply by flexa-david** [reply]
Works perfectly. One of those "why didn't I think of that" moments! Thanks so much for taking the time to code that out!
